### PR TITLE
feat: add an accounts api to simplify access for commonly used accounts

### PR DESCRIPTION
**Pull request by aaronschweig**
This PR adds a service that exposes the following interface to allow consumers to interact with commonly used methods of loading specific accounts:

```go
GetFirstLevelAccountForAccount(ctx context.Context, accountKey client.ObjectKey) (*Account, error)
GetFirstLevelAccountForNamespace(ctx context.Context, namespace string) (*Account, error)

GetAccount(ctx context.Context, accountKey client.ObjectKey) (*Account, error)
GetAccountForNamespace(ctx context.Context, namespace string) (*Account, error)
```

It also includes a set of tests to ensure that the implemented logic is working properly

### CODE REVIEW

File: api/v1alpha1/service.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+func NewService(client client.Client, rootNamespace string) *service {
+	return &service{
+		client:        client,
+		rootNamespace: rootNamespace,
+	}
+}
+
+func (s *service) GetFirstLevelAccountForAccount(ctx context.Context, accountKey client.ObjectKey) (*Account, error) {
+	return s.GetAccountForNamespace(ctx, accountKey.Namespace)
```
Comment (nexus49) [inline api/v1alpha1/service.go:34]: This should be GetFirstLevelAccountForNamespace right?

File: api/v1alpha1/service.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func (s *service) GetFirstLevelAccountForNamespace(ctx context.Context, namespace string) (*Account, error) {
+	var ns corev1.Namespace
```
Comment (nexus49) [inline api/v1alpha1/service.go:38]: should we create a helper function for this ns lookup? Seems this is repeated below

**Review by nexus49 — CHANGES_REQUESTED** [review]

File: api/v1alpha1/service.go (hunk 1 above)
Comment (aaronschweig) [reply api/v1alpha1/service.go:34]: yes you are right - I changed it!

File: api/v1alpha1/service.go (hunk 2 above)
Comment (aaronschweig) [reply api/v1alpha1/service.go:38]: good idea - i refactored it and put it into its own helper function

**Review by tobias-oetzel — APPROVED** [review]

**Review by nexus49 — APPROVED** [review]

**Pull request merged**